MapboxNavigation/InstructionLabel.swift

```diff
@@ -28,10 +29,28 @@ open class InstructionLabel: StylableLabel, InstructionPresenterDataSource {
             
             let presenter = InstructionPresenter(instruction, dataSource: self, imageRepository: imageRepository, downloadCompletion: update)
             
-            attributedText = presenter.attributedText()
+            let attributed = presenter.attributedText()
+            attributedText = instructionDelegate?.label?(self, willPresent: instruction, as: attributed) ?? attributed
             instructionPresenter = presenter
         }
     }
 
     private var instructionPresenter: InstructionPresenter?
 }
+
+/**
+ The `VoiceControllerDelegate` protocol defines a method that allows an object to customize presented visual instructions.
+ */
+@objc(MBVisualInstructionDelegate)
+public protocol VisualInstructionDelegate: class {
+    
+    /**
+     Called when an InstructionLabel will present a visual instruction.
+     
+     - parameter label: The label that the instruction will be presented on.
+     - parameter instruction: the `VisualInstruction` that will be presented.
+     - parameter presented: the formatted string that is provided by the instruction presenter
+     - returns: optionally, a customized NSAttributedString that will be presented instead of the default, or if nil, the default behavior will be used.
+     */
+    @objc optional func label(_ label: InstructionLabel, willPresent instruction: VisualInstruction, as presented: NSAttributedString) -> NSAttributedString?
```

Comment:
A delegate method’s first argument should always be the delegator. Is the delegator an InstructionLabel, or something else? Can it be either or two classes? That would be an antipattern in Cocoa.

Does this “will” method’s name make the implementation’s purpose clear enough? Is it clear that it’s an opportunity to override the presentation?